src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -687,19 +693,47 @@ private void WriteXmpChunk(Stream stream, ImageMetadata meta)
                 PngConstants.XmpKeyword.CopyTo(payload);
                 int bytesWritten = PngConstants.XmpKeyword.Length;
 
-                // Write the iTxt header (all zeros in this case)
+                // Write the iTxt header (all zeros in this case).
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
                 payload[bytesWritten++] = 0;
 
-                // And the XMP data itself
+                // And the XMP data itself.
                 xmpData.CopyTo(payload.Slice(bytesWritten));
                 this.WriteChunk(stream, PngChunkType.InternationalText, payload);
             }
         }
 
+        /// <summary>
+        /// Writes the color profile chunk.
+        /// </summary>
+        /// <param name="stream">The stream to write to.</param>
+        /// <param name="metaData">The image meta data.</param>
+        private void WriteColorProfileChunk(Stream stream, ImageMetadata metaData)
+        {
+            if (metaData.IccProfile is null)
+            {
+                return;
+            }
+
+            byte[] iccProfileBytes = metaData.IccProfile.ToByteArray();
+
+            byte[] compressedData = this.GetZlibCompressedBytes(iccProfileBytes);
+            int payloadLength = ColorProfileName.Length + compressedData.Length + 2;
+            using (IMemoryOwner<byte> owner = this.memoryAllocator.Allocate<byte>(payloadLength))
+            {
+                Span<byte> outputBytes = owner.GetSpan();
+                PngConstants.Encoding.GetBytes(ColorProfileName).CopyTo(outputBytes);
+                int bytesWritten = ColorProfileName.Length;
+                outputBytes[bytesWritten++] = 0; // Null separator.
+                outputBytes[bytesWritten++] = 0; // Compression.
```

Comment:
Here the above "trick" doesn't make sense, as for slicing there's an argument validation, so we just would exchange one comparison for the bound-check against the argument validation. So no net-win.